scipy/sparse/linalg/_isolve/symmlq.py

```diff
@@ -0,0 +1,239 @@
+import numpy as np
+from .utils import make_system
+
+
+__all__ = ['symmlq']
+
+
+def symmlq(A, b, x0=None, tol=1e-5, maxiter=None, M=None, callback=None,
```

Comment:
> tolerance names are very cryptic. 

I'll modify as
```python
def symmlq(A, b, x0=None, tol=1e-5, atol=None, maxiter=None, M=None, callback=None)
```
and clarify description of `tol` and `atol` in the docstring.

> Also haptol is currently less than the machine precision for floats which is ~2.2e-16 so practically 0.

The `haptol` is set to $1e-18$ by default in PETSc, I chose the same setting in SciPy. Let me give an example to explain why `haptol` is not set to $0.$ in implementation. Firstly, adding `print(f"abs(dp) = {abs(dp)}")` on line 143 in `symmlq.py`, and then:
```python
import numpy as np
from scipy.sparse import csr_matrix
from symmlq import *


A = csr_matrix([[2, -1, 0], [-1, 2, -1], [0, -1, 2]], dtype=float)
b = np.ones(A.shape[0])
eps = 1e-10
x0 = np.array([1.5, 2., 1.5+eps])
x, info = symmlq(A, b, x0=x0, tol=eps, show=True)
```
you can find the output of `abs(dp)` is `abs(dp) = 5.000000827403744e-20` (where $dp := r^TMr$, $M$ is an identity, $r:=b - Ax$), which is not zero ($> 0.$) but less than $1e-18$, so we believe SYMMLQ achieves happy breakdown in the sense of $1e-18$.

Reply:
The relative tolerance should read `rtol` not `tol` still not the best naming but at least rtol is widely known. You can't force something to be less than machine precision with 64-bit float precision. That is impossible. So if it is used as a scalar somewhere it's another story. I didn't read the paper or the code but otherwise this makes no sense.

Reply:
> The relative tolerance should read `rtol` not `tol` still not the best naming but at least rtol is widely known. 

Ok, this parameter name `tol` will be changed as `rtol` in the next submission. But before that, I'll firstly submit other PR to change its name because `tol` largely exists in the file `tests/test_iterative.py`, which will fail the test cases extended to `symmlq` in `test_iterative.py`.

> So if it is used as a scalar somewhere it's another story.

Yes, it is used as a scalar, it largely exists in PDEs numerical computing, such as punishment method in methods, drift-diffusion equations for protein ion channel in physical models.

Reply:
> So if it is used as a scalar somewhere it's another story.

Yes, it is used as a scalar. This situation can be encountered in PDE numerical computing such as punishment method.